packages/block-editor/src/hooks/border-width.js

```diff
@@ -24,27 +25,40 @@ export const BorderWidthEdit = ( props ) => {
 		setAttributes,
 	} = props;
 
+	// Step value is maintained in state so step is appropriate for current unit
+	// even when current radius value is undefined.
+	const initialStep = parseUnit( style?.border?.width ) === 'px' ? 1 : 0.25;
+	const [ step, setStep ] = useState( initialStep );
+
+	const onUnitChange = ( newUnit ) => {
+		setStep( newUnit === 'px' ? 1 : 0.25 );
+	};
+
 	const onChange = ( newWidth ) => {
-		const newStyle = {
+		let newStyle = {
 			...style,
 			border: {
 				...style?.border,
 				width: newWidth,
 			},
 		};
 
+		if ( newWidth === undefined || newWidth === '' ) {
```

Comment:
As per a few lines below, we already clean the object in all cases, why do we need this?

Reply:
Or rather: why do we clean the object? Is it to remove the width when it has a falsy value? If so, we don't need to do it always.

Reply:
Thanks for catching that one. An oversight while resolving a conflict during a rebase.

The intention is to allow the `0` border width value. I've removed the erroneous call to `cleanEmptyObject`.